src/GameLogic/PlayerActions/Craftings/FenrirUpgradeCrafting.cs

```diff
@@ -31,16 +31,15 @@ protected override int GetPrice(byte successRate, IList<CraftingRequiredItemLink
         var inputItems = player.TemporaryStorage!.Items.ToList();
         var itemsLevelAndOption4 = inputItems
             .Where(item => item.Level >= 4
-                           && item.ItemOptions.Any(o => o.ItemOption?.OptionType == ItemOptionTypes.Option))
+                && item.ItemOptions.Any(o => o.ItemOption?.OptionType == ItemOptionTypes.Option))
             .ToList();
         var randomWeapons = itemsLevelAndOption4
             .Where(item => item.IsWearable()
-                           && item.Definition!.BasePowerUpAttributes.Any(a =>
-                               a.TargetAttribute == Stats.MaximumPhysBaseDmgByWeapon))
+                && item.Definition!.BasePowerUpAttributes.Any(a => a.TargetAttribute == Stats.AttackSpeedByWeapon))
```

Comment:
Summoner's books have 0 physical damage. `AttackSpeedByWeapon` is now the common attribute shared by all weapons.